src/core/ipfs/files.js

```diff
@@ -3,19 +3,20 @@
 const Importer = require('ipfs-unixfs-engine').importer
 const Exporter = require('ipfs-unixfs-engine').exporter
 const UnixFS = require('ipfs-unixfs')
+const promisify = require('promisify-es6')
 
 module.exports = function files (self) {
   return {
-    add: (arr, callback) => {
+    add: promisify((arr, cb) => {
```

Comment:
our pattern is to use `callback` as the most top level callback, so that it is easy distinguishable when we jump out the func